GunShipAlpha/src/Projectile.java

```diff
@@ -0,0 +1,33 @@
+import java.awt.*;
+import java.awt.geom.Point2D;
+
+class Projectile {
+    private double x, y;
+   private final double dx;
+    private final double dy;
```

Comment:
dx and dy together with a point is called a vector. basically "where does it start, what direction, and how fast". you can also store it as point, angle, speed. here your projectile is _almost_ a vector, but could definitely contain a vector (eventually).